components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -69,6 +69,8 @@
     private static final String OAUTH_VERSION = "OAuth-2.0";
     private static final String GRANT_TYPE_SEPARATOR = " ";
     private static Pattern clientIdRegexPattern = null;
+    private static final String[] TOKEN_ENDPOINT_AUTH_METHODS = {"none", "client_secret_post",
+            "client_secret_basic", "private_key_jwt"};
```

Comment:
need to include mutual_tls

Reply:
Also token endpoint auth methods should not be hard-coded.. IMO we should make it configurable or need to capture them dynamically when authenticators are registered